### ISSUE TITLE
Scan only-testing flag mishandles arrays of strings, causing build to fail

### ISSUE THREAD

**Issue by AaronSofaer** [fastlane/fastlane#8342]
### New Issue Checklist

- [X] Updated fastlane to the latest version
- [X] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description
Fastlane Scan's documentation gives this argument option, matching xcodebuild's -only-testing argument:

> | only_testing            | Array of strings matching Test Bundle/Test Suite/Test Cases to run  

When passing an array of strings to this which I have composed out of all the tests that have failed, the resultant xcodebuild output doesn't seem right, and it definitely throws an error.

If I'm doing something wrong, happy to learn that, because the UI Tests are sufficiently unreliable that I was really really looking forward to being able to retest failures.

This StackOverflow post suggests to me that perhaps there are quotes missing from the xcodebuild invocation. http://stackoverflow.com/questions/39427263/how-to-use-xcodebuild-with-only-testing-and-skip-testing-flag

##### Complete output when running fastlane, including the stack trace and command used
> You can use: `--capture_output` as the last commandline argument to get that collected for you

`[11:53:28]: [4m[36m$ set -o pipefail && env NSUnbufferedIO=YES xcodebuild -workspace PlanGrid.xcworkspace -scheme PlanGrid_UI_Tests -destination 'platform=iOS Simulator,id=6F60DBEB-2E43-4D9B-8536-8FD92F092EA9' -derivedDataPath 'DerivedData' -enableCodeCoverage YES -only-testing:PlanGridUITests/AttachmentUITests/testSampleProjectAttachments -only-testing:PlanGridUITests/AuthUITests/testAuthFailures -only-testing:PlanGridUITests/AuthUITests/testSignUpWithFields -only-testing: -only-testing: -only-testing: -only-testing:PlanGridUITests/IssuesLogUITests/testCreateAndBatchEditIssues -only-testing:PlanGridUITests/IssuesLogUITests/testFixtureIssueLog -only-testing:PlanGridUITests/IssuesLogUITests/testIssueLogSearch -only-testing: -only-testing:PlanGridUITests/IssuesLogUITests/testSingleIssueChanges -only-testing: -only-testing:PlanGridUITests/ProjectsUITests/testRemoveProjectFromDevice -only-testing:PlanGridUITests/ProjectsUITests/testStorageManagement -only-testing: -only-testing:PlanGridUITests/RFIUITests/testRFICreation -only-testing: -only-testing:PlanGridUITests/SheetViewUITests/testAddMasterDeleteAnnotations -only-testing:PlanGridUITests/SheetViewUITests/testCollaboratorPermissions -only-testing: -only-testing:PlanGridUITests/SheetViewUITests/testPowerCollaboratorPermissions -only-testing:PlanGridUITests/SheetViewUITests/testRegressions -only-testing:PlanGridUITests/SheetViewUITests/testRulerCalibrationDialog -only-testing: -only-testing:PlanGridUITests/SnapshotsUITests/testCreateThenZoomSnapshot -only-testing: test-without-building | tee '/Users/jenkins/Library/Logs/scan/Pods_PlanLib-PlanGrid_UI_Tests.log' | xcpretty [0m[0m`

```[11:53:28]: ▸ [35mLoading...[0m
[11:53:28]: ▸ [35mxcodebuild: error: invalid option '-only-testing:'[0m
[11:53:28]: ▸ [35mUsage: xcodebuild [-project <projectname>] [[-target <targetname>]...|-alltargets] [-configuration <configurationname>] [-arch <architecture>]... [-sdk [<sdkname>|<sdkpath>]] [-showBuildSettings] [<buildsetting>=<value>]... [<buildaction>]...[0m
[11:53:28]: ▸ [35mxcodebuild [-project <projectname>] -scheme <schemeName> [-destination <destinationspecifier>]... [-configuration <configurationname>] [-arch <architecture>]... [-sdk [<sdkname>|<sdkpath>]] [-showBuildSettings] [<buildsetting>=<value>]... [<buildaction>]...[0m
[11:53:28]: ▸ [35mxcodebuild -workspace <workspacename> -scheme <schemeName> [-destination <destinationspecifier>]... [-configuration <configurationname>] [-arch <architecture>]... [-sdk [<sdkname>|<sdkpath>]] [-showBuildSettings] [<buildsetting>=<value>]... [<buildaction>]...[0m
[11:53:28]: ▸ [35mxcodebuild -version [-sdk [<sdkfullpath>|<sdkname>] [<infoitem>] ][0m
[11:53:28]: ▸ [35mxcodebuild -list [[-project <projectname>]|[-workspace <workspacename>]] [-json][0m
[11:53:28]: ▸ [35mxcodebuild -showsdks[0m
[11:53:28]: ▸ [35mxcodebuild -exportArchive -archivePath <xcarchivepath> -exportPath <destinationpath> -exportOptionsPlist <plistpath>[0m
[11:53:28]: ▸ [35mxcodebuild -exportLocalizations -localizationPath <path> -project <projectname> [-exportLanguage <targetlanguage>...][0m
[11:53:28]: ▸ [35mxcodebuild -importLocalizations -localizationPath <path> -project <projectname>[0m
[11:53:28]: ▸ [35mOptions:[0m
[11:53:28]: ▸ [35m-usage                              print brief usage[0m
[11:53:28]: ▸ [35m-help                               print complete usage[0m
[11:53:28]: ▸ [35m-verbose                            provide additional status output[0m
[11:53:28]: ▸ [35m-license                            show the Xcode and SDK license agreements[0m
[11:53:28]: ▸ [35m-checkFirstLaunchStatus             Check if any First Launch tasks need to be performed[0m
[11:53:28]: ▸ [35m-project NAME                       build the project NAME[0m
[11:53:28]: ▸ [35m-target NAME                        build the target NAME[0m
[11:53:28]: ▸ [35m-alltargets                         build all targets[0m
[11:53:28]: ▸ [35m-workspace NAME                     build the workspace NAME[0m
[11:53:28]: ▸ [35m-scheme NAME                        build the scheme NAME[0m
[11:53:28]: ▸ [35m-configuration NAME                 use the build configuration NAME for building each target[0m
[11:53:28]: ▸ [35m-xcconfig PATH                      apply the build settings defined in the file at PATH as overrides[0m
[11:53:28]: ▸ [35m-arch ARCH                          build each target for the architecture ARCH; this will override architectures defined in the project[0m
[11:53:28]: ▸ [35m-sdk SDK                            use SDK as the name or path of the base SDK when building the project[0m
[11:53:28]: ▸ [35m-toolchain NAME                     use the toolchain with identifier or name NAME[0m
[11:53:28]: ▸ [35m-destination DESTINATIONSPECIFIER   use the destination described by DESTINATIONSPECIFIER (a comma-separated set of key=value pairs describing the destination to use)[0m
[11:53:28]: ▸ [35m-destination-timeout TIMEOUT        wait for TIMEOUT seconds while searching for the destination device[0m
[11:53:28]: ▸ [35m-parallelizeTargets                 build independent targets in parallel[0m
[11:53:28]: ▸ [35m-jobs NUMBER                        specify the maximum number of concurrent build operations[0m
[11:53:28]: ▸ [35m-dry-run                            do everything except actually running the commands[0m
[11:53:28]: ▸ [35m-quiet                              do not print any output except for warnings and errors[0m
[11:53:28]: ▸ [35m-hideShellScriptEnvironment         don't show shell script environment variables in build log[0m
[11:53:28]: ▸ [35m-showsdks                           display a compact list of the installed SDKs[0m
[11:53:28]: ▸ [35m-showBuildSettings                  display a list of build settings and values[0m
[11:53:28]: ▸ [35m-list                               lists the targets and configurations in a project, or the schemes in a workspace[0m
[11:53:28]: ▸ [35m-find-executable NAME               display the full path to executable NAME in the provided SDK and toolchain[0m
[11:53:28]: ▸ [35m-find-library NAME                  display the full path to library NAME in the provided SDK and toolchain[0m
[11:53:28]: ▸ [35m-version                            display the version of Xcode; with -sdk will display info about one or all installed SDKs[0m
[11:53:28]: ▸ [35m-enableAddressSanitizer YES|NO      turn the address sanitizer on or off[0m
[11:53:28]: ▸ [35m-enableThreadSanitizer YES|NO       turn the thread sanitizer on or off[0m
[11:53:28]: ▸ [35m-resultBundlePath PATH              specifies the directory where a result bundle describing what occurred will be placed[0m
[11:53:28]: ▸ [35m-derivedDataPath PATH               specifies the directory where build products and other derived data will go[0m
[11:53:28]: ▸ [35m-archivePath PATH                   specifies the directory where any created archives will be placed, or the archive that should be exported[0m
[11:53:28]: ▸ [35m-exportArchive                      specifies that an archive should be exported[0m
[11:53:28]: ▸ [35m-exportOptionsPlist PATH            specifies a path to a plist file that configures archive exporting[0m
[11:53:28]: ▸ [35m-enableCodeCoverage YES|NO          turn code coverage on or off when testing[0m
[11:53:28]: ▸ [35m-exportPath PATH                    specifies the destination for the product exported from an archive[0m
[11:53:28]: ▸ [35m-skipUnavailableActions             specifies that scheme actions that cannot be performed should be skipped instead of causing a failure[0m
[11:53:28]: ▸ [35m-exportLocalizations                exports completed and outstanding project localizations[0m
[11:53:28]: ▸ [35m-importLocalizations                imports localizations for a project, assuming any necessary localized resources have been created in Xcode[0m
[11:53:28]: ▸ [35m-localizationPath                   specifies a path to XLIFF localization files[0m
[11:53:28]: ▸ [35m-exportLanguage                     specifies multiple optional ISO 639-1 languages included in a localization export[0m
[11:53:28]: ▸ [35m-xctestrun                          specifies a path to a test run specification[0m
[11:53:28]: ▸ [35m-only-testing:TEST-IDENTIFIER       constrains testing by specifying tests to include, and excluding other tests[0m
[11:53:28]: ▸ [35m-skip-testing:TEST-IDENTIFIER       constrains testing by specifying tests to exclude, but including other tests[0m
[11:53:28]: ▸ [35m-json                               output as JSON (note: -json implies -quiet)[0m
xcodebuild: error: invalid option '-only-testing:'

Usage: xcodebuild [-project <projectname>] [[-target <targetname>]...|-alltargets] [-configuration <configurationname>] [-arch <architecture>]... [-sdk [<sdkname>|<sdkpath>]] [-showBuildSettings] [<buildsetting>=<value>]... [<buildaction>]...
xcodebuild [-project <projectname>] -scheme <schemeName> [-destination <destinationspecifier>]... [-configuration <configurationname>] [-arch <architecture>]... [-sdk [<sdkname>|<sdkpath>]] [-showBuildSettings] [<buildsetting>=<value>]... [<buildaction>]...
xcodebuild -workspace <workspacename> -scheme <schemeName> [-destination <destinationspecifier>]... [-configuration <configurationname>] [-arch <architecture>]... [-sdk [<sdkname>|<sdkpath>]] [-showBuildSettings] [<buildsetting>=<value>]... [<buildaction>]...
xcodebuild -version [-sdk [<sdkfullpath>|<sdkname>] [<infoitem>] ]
xcodebuild -list [[-project <projectname>]|[-workspace <workspacename>]] [-json]
xcodebuild -showsdks
xcodebuild -exportArchive -archivePath <xcarchivepath> -exportPath <destinationpath> -exportOptionsPlist <plistpath>
xcodebuild -exportLocalizations -localizationPath <path> -project <projectname> [-exportLanguage <targetlanguage>...]
xcodebuild -importLocalizations -localizationPath <path> -project <projectname>

Options:
-usage                              print brief usage
-help                               print complete usage
-verbose                            provide additional status output
-license                            show the Xcode and SDK license agreements
-checkFirstLaunchStatus             Check if any First Launch tasks need to be performed
-project NAME                       build the project NAME
-target NAME                        build the target NAME
-alltargets                         build all targets
-workspace NAME                     build the workspace NAME
-scheme NAME                        build the scheme NAME
-configuration NAME                 use the build configuration NAME for building each target
-xcconfig PATH                      apply the build settings defined in the file at PATH as overrides
-arch ARCH                          build each target for the architecture ARCH; this will override architectures defined in the project
-sdk SDK                            use SDK as the name or path of the base SDK when building the project
-toolchain NAME                     use the toolchain with identifier or name NAME
-destination DESTINATIONSPECIFIER   use the destination described by DESTINATIONSPECIFIER (a comma-separated set of key=value pairs describing the destination to use)
-destination-timeout TIMEOUT        wait for TIMEOUT seconds while searching for the destination device
-parallelizeTargets                 build independent targets in parallel
-jobs NUMBER                        specify the maximum number of concurrent build operations
-dry-run                            do everything except actually running the commands
-quiet                              do not print any output except for warnings and errors
-hideShellScriptEnvironment         don't show shell script environment variables in build log
-showsdks                           display a compact list of the installed SDKs
-showBuildSettings                  display a list of build settings and values
-list                               lists the targets and configurations in a project, or the schemes in a workspace
-find-executable NAME               display the full path to executable NAME in the provided SDK and toolchain
-find-library NAME                  display the full path to library NAME in the provided SDK and toolchain
-version                            display the version of Xcode; with -sdk will display info about one or all installed SDKs
-enableAddressSanitizer YES|NO      turn the address sanitizer on or off
-enableThreadSanitizer YES|NO       turn the thread sanitizer on or off
-resultBundlePath PATH              specifies the directory where a result bundle describing what occurred will be placed
-derivedDataPath PATH               specifies the directory where build products and other derived data will go
-archivePath PATH                   specifies the directory where any created archives will be placed, or the archive that should be exported
-exportArchive                      specifies that an archive should be exported
-exportOptionsPlist PATH            specifies a path to a plist file that configures archive exporting
-enableCodeCoverage YES|NO          turn code coverage on or off when testing
-exportPath PATH                    specifies the destination for the product exported from an archive
-skipUnavailableActions             specifies that scheme actions that cannot be performed should be skipped instead of causing a failure
-exportLocalizations                exports completed and outstanding project localizations
-importLocalizations                imports localizations for a project, assuming any necessary localized resources have been created in Xcode
-localizationPath                   specifies a path to XLIFF localization files
-exportLanguage                     specifies multiple optional ISO 639-1 languages included in a localization export
-xctestrun                          specifies a path to a test run specification
-only-testing:TEST-IDENTIFIER       constrains testing by specifying tests to include, and excluding other tests
-skip-testing:TEST-IDENTIFIER       constrains testing by specifying tests to exclude, but including other tests
-json                               output as JSON (note: -json implies -quiet)
[11:53:28]: [31mExit status: 64[0m
+----------------------------+-------------+
|               [33mLane Context[0m               |
+----------------------------+-------------+
| DEFAULT_PLATFORM           | ios         |
| PLATFORM_NAME              | ios         |
| LANE_NAME                  | ios uitest  |
| SCAN_DERIVED_DATA_PATH     | DerivedData |
| SCAN_GENERATED_PLIST_FILES | []          |
| SCAN_GENERATED_PLIST_FILE  |             |
+----------------------------+-------------+
[11:53:28]: [31mError building/testing the application - see the log above[0m```

**Comment by AaronSofaer** [comment]
Looks like I've got some empty strings in there somehow, which caused the problem. Whoops. Closing.
